### ISSUE TITLE
NPM testing generates a number of warnings and errors

### ISSUE THREAD

**Issue by kallewesterling** [alan-turing-institute/AssurancePlatform#357]
NPM testing generates a number of warnings and errors:

```
> frontend@0.1.0 test
> jest

Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
(node:60853) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 PASS  src/components/tests/MermaidSyntax.test.js
(node:60855) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 PASS  src/components/tests/Mermaid.test.js (5.804 s)
(node:60861) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 PASS  src/components/tests/CreateGroup.test.js (10.935 s)
(node:60860) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 PASS  src/components/tests/Logout.test.js (10.954 s)
(node:60856) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 PASS  src/components/tests/CaseCreator.test.js (6.489 s)
 PASS  src/components/tests/Groups.test.js (11.248 s)
(node:60859) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:60858) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 PASS  src/components/tests/ParentSelector.test.js (11.671 s)
  ● Console

    console.error
      Warning: An update to ParentSelector inside a test was not wrapped in act(...).
      
      When testing, code that causes React state updates should be wrapped into act(...):
      
      act(() => {
        /* fire events that update state */
      });
      /* assert on the output */
      
      This ensures that you're testing the behavior the user would see in the browser. Learn more at https://reactjs.org/link/wrap-tests-with-act
          at ParentSelector (/Users/kwesterling/Repositories/AssurancePlatform/frontend/src/components/ParentSelector.js:7:3)

      35 |
      36 |         if (!potential) {
    > 37 |           setOptions(currentParents);
         |           ^
      38 |         } else {
      39 |           let potentialParents = await itemGetPotentialParents(
      40 |             token,

      at printWarning (node_modules/react-dom/cjs/react-dom.development.js:67:30)
      at error (node_modules/react-dom/cjs/react-dom.development.js:43:5)
      at warnIfNotCurrentlyActingUpdatesInDEV (node_modules/react-dom/cjs/react-dom.development.js:24064:9)
      at setOptions (node_modules/react-dom/cjs/react-dom.development.js:16135:9)
      at _callee$ (src/components/ParentSelector.js:37:11)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:64:40)
      at Generator.invoke (node_modules/regenerator-runtime/runtime.js:299:22)
      at Generator.next (node_modules/regenerator-runtime/runtime.js:124:21)

    console.error
      Warning: An update to ParentSelector inside a test was not wrapped in act(...).
      
      When testing, code that causes React state updates should be wrapped into act(...):
      
      act(() => {
        /* fire events that update state */
      });
      /* assert on the output */
      
      This ensures that you're testing the behavior the user would see in the browser. Learn more at https://reactjs.org/link/wrap-tests-with-act
          at ParentSelector (/Users/kwesterling/Repositories/AssurancePlatform/frontend/src/components/ParentSelector.js:7:3)

      52 |             return true;
      53 |           });
    > 54 |           setOptions(potentialParents);
         |           ^
      55 |         }
      56 |       })
      57 |       .catch((err) => {

      at printWarning (node_modules/react-dom/cjs/react-dom.development.js:67:30)
      at error (node_modules/react-dom/cjs/react-dom.development.js:43:5)
      at warnIfNotCurrentlyActingUpdatesInDEV (node_modules/react-dom/cjs/react-dom.development.js:24064:9)
      at setOptions (node_modules/react-dom/cjs/react-dom.development.js:16135:9)
      at _callee$ (src/components/ParentSelector.js:54:11)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:64:40)
      at Generator.invoke (node_modules/regenerator-runtime/runtime.js:299:22)
      at Generator.next (node_modules/regenerator-runtime/runtime.js:124:21)

 PASS  src/components/tests/ItemEditor.test.js (11.707 s)
(node:60857) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 PASS  src/components/tests/Signup.test.js (11.744 s)
(node:60854) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 PASS  src/components/tests/Login.test.js (11.844 s)

Test Suites: 10 passed, 10 total
Tests:       34 passed, 34 total
Snapshots:   0 total
Time:        15.859 s
Ran all test suites.
```

**Comment by edchapman88** [comment]
closed with 98b5c26cef599c4755a9c21807f9e3fc223dfe2a.

Note: this commit also **removes** the `Groups` tests because they tested the `Groups` component which is currently not used in the app and has been commented out (awaiting a refactor to remove the grommet components if this needs to be re-introduced into the app at a later date).

Since the whole `Groups.test.js` file was removed, it can easily be reinstated at a later date by checking it out in the commit history. (eg. `git checkout <a commit hash with the file in> "frontend/src/components/tests/Groups.test.js"`)

**Comment by kallewesterling** [comment]
Thank you for this @edchapman88!! While not a breaking thing, this will make the experience a lot smoother for anyone trying to run the package locally :)
